tests/ops/op_math/test_adjoint_op.py

```diff
@@ -0,0 +1,587 @@
+        op = Adjoint(qml.PauliX(0) @ qml.PauliY(1))
+        assert op._queue_category is None
+
+
+class TestMiscMethods:
+    """Test miscellaneous small methods on the Adjoint class."""
+
+    def test_label(self):
+        """Test that the label method for the adjoint class adds a † to the end."""
+        base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+        op = Adjoint(base)
+        assert op.label(decimals=2) == "Rot\n(1.23,\n2.35,\n3.46)†"
+
+    def test_adjoint_of_adjoint(self):
+        """Test that the adjoint of an adjoint is the original operation."""
+        base = qml.PauliX(0)
+        op = Adjoint(base)
+
+        assert op.adjoint() is base
+
+    def test_diagonalizing_gates(self):
+        """Assert that the diagonalizing gates method gives the base's diagonalizing gates."""
+        base = qml.Hadamard(0)
+        diag_gate = Adjoint(base).diagonalizing_gates()[0]
+
+        assert isinstance(diag_gate, qml.RY)
+        assert qml.math.allclose(diag_gate.data[0], -np.pi / 4)
+
+
+class TestAdjointOperation:
+    """Test methods in the AdjointOperation mixin."""
+
+    @pytest.mark.parametrize(
+        "base, adjoint_base_name",
+        ((qml.PauliX(0), "Adjoint(PauliX)"), (qml.RX(1.2, wires=0), "Adjoint(RX)")),
+    )
+    def test_base_name(self, base, adjoint_base_name):
+        """Test the base_name property of AdjointOperation."""
+        op = Adjoint(base)
+        assert op.base_name == adjoint_base_name
+
+    def test_generator(self):
+        """Assert that the generator of an Adjoint is -1.0 times the base generator."""
+        base = qml.RX(1.23, wires=0)
+        op = Adjoint(base)
+
+        assert base.generator().compare(-1.0 * op.generator())
+
+    def test_no_generator(self):
+        """Test that an adjointed non-Operation raises a GeneratorUndefinedError."""
+
+        with pytest.raises(qml.operation.GeneratorUndefinedError):
+            Adjoint(1.0 * qml.PauliX(0)).generator()
+
+    def test_single_qubit_rot_angles(self):
+
+        param = 1.234
+        base = qml.RX(param, wires=0)
+        op = Adjoint(base)
+
+        base_angles = base.single_qubit_rot_angles()
+        angles = op.single_qubit_rot_angles()
+
+        for angle1, angle2 in zip(angles, reversed(base_angles)):
+            assert angle1 == -angle2
+
+    @pytest.mark.parametrize(
+        "base, basis",
+        (
+            (qml.RX(1.234, wires=0), "X"),
+            (qml.PauliY("a"), "Y"),
+            (qml.PhaseShift(4.56, wires="b"), "Z"),
+            (qml.SX(-1), "X"),
+        ),
+    )
+    def test_basis_property(self, base, basis):
+        op = Adjoint(base)
+        assert op.basis == basis
+
+    def test_control_wires(self):
+        """Test the control_wires of an adjoint are the same as the base op."""
+        op = Adjoint(qml.CNOT(wires=("a", "b")))
+        assert op.control_wires == qml.wires.Wires("a")
+
+
+class TestInverse:
+    """Tests involving the inverse attribute."""
```

Comment:
Those tests can be deleted soon?

Reply:
Probably not soon since we should have a long deprecation cycle.